### PR TITLE
README updates

### PR DESCRIPTION
**Pull request by Yasumoto**
📖   Small corrections, updates to the latest API, and grammar tweaks.

### Motivation:

Make sure our README is current and easy to follow.

### Modifications:

1. Graphite is a backend for metrics, Grafana is a frontend for visualization+alerting that can be powered by many backends.
1. We have a tagged release.
1. Add a link to @mrlotu 's SwiftPrometheus (emoji optional 😄 )
1. Some (mostly optional) punctuation tweaks
1. Tried to re-word the "applications choose their backend" section so it does not use the passive voice.
1. Update to the latest API... do we still want to write this out this here vs. link to the auto-generated API docs? 🤔 I'm leaning toward deleting it, but there's some value in it all being present here so long as we remember to update it.

### Result:

Ideally, a better README! 📖

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,20 +55,21 @@ This instructs the `MetricsSystem` to install `SelectedMetricsImplementation` (a
 
 As the API has just launched, not many implementations exist yet. If you are interested in implementing one see the "Implementing a metrics backend" section below explaining how to do so. List of existing SwiftMetrics API compatible libraries:
 
+- 🔥 [SwiftPrometheus](https://github.com/MrLotU/SwiftPrometheus), support for [Prometheus](https://prometheus.io)
```
Comment (Lukasa) [inline README.md]: I think it might be best to remove the emoji here: sorry! 😆

File: README.md [hunk 2]
```diff
@@ -110,13 +109,14 @@ timer.recordMilliseconds(100)
 
 Note: Unless you need to implement a custom metrics backend, everything in this section is likely not relevant, so please feel free to skip.
 
-As seen above, each of `Counter`, `Timer`, `Recorder` and `Gauge` constructors provides a metric object. This raises the question of which metrics backend is actually be used when calling these constructors? The answer is that it's configurable _per application_. The application sets up the metrics backend it wishes to use. Configuring the metrics backend is straightforward:
+As seen above, each contstructor for `Counter`, `Timer`, `Recorder` and `Gauge` provides a metric object. This uncertainty obscures the selected metrics backend calling these constructors by design. _Each application_ can select and configure its desired backend. The application sets up the metrics backend it wishes to use. Configuring the metrics backend is straightforward:
```
Comment (Lukasa) [inline README.md]: Typo in "constructor"

Comment (Yasumoto) [reply README.md]: Great save, fixed!

**Comment by tomerd** [comment]
@swift-server-bot test this please

**Comment by tomerd** [comment]
thanks @Yasumoto /cc

**Pull request merged**